### PR TITLE
Adds Support for Moonbeam, and Moonriver

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Adds Support for Moonbeam, and Moonriver.

## How to test

Same way as other EVM chains

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: registry.json [hunk 1]
```diff
@@ -2487,5 +2487,88 @@
...
+    },
+    "info": {
+      "url": "https://moonbeam.network",
+      "rpc": "https://rpc.api.moonbeam.network",
+      "documentation": "https://docs.moonbeam.network"
+    }
+  },
+  {
+    "id": "bittorrent",
+    "name": "BitTorrent Chain",
```
Comment (hewigovens) [inline registry.json:2548]: moonriver and moonbeam looks legit, how about this one?

**Comment by AyushBherwani1998** [comment]
@hewigovens Do I need to add tests for new chains? If it's required I can add. I'll update the assets repository soon.

File: registry.json (hunk 1 above)
Comment (AyushBherwani1998) [reply registry.json:2548]: Let me get back to you, I'll fix the failing test once we confirm if we should add BitTorrent or not.

**Comment by hewigovens** [comment]
@AyushBherwani1998 have you tried `codegen/bin/newcoin` command?

**Comment by AyushBherwani1998** [comment]
> @AyushBherwani1998 have you tried `codegen/bin/newcoin` command?

Yes, I have tried the command, but it adds files which are not required for the EVM chain support. While adding the EVM support I only use `tools/generate-files` and `codegen/bin/cointests`. Should I prefer using that instead?

File: registry.json (hunk 1 above)
Comment (AyushBherwani1998) [reply registry.json]: Have decided to skip BitTorrent chain.

**Comment by AyushBherwani1998** [comment]
Relevant assets PR: https://github.com/trustwallet/assets/pull/20876

**Review by hewigovens — APPROVED** [review]

File: docs/registry.md [hunk 2]
```diff
@@ -78,6 +78,8 @@ This list is generated from [./registry.json](../registry.json)
 | 10000321 | KuCoin Community Chain | KCS    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/kcc/info/logo.png" width="32" />          | <https://www.kcc.io/>         |
 | 10000553 | Huobi ECO Chain  | HT     | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/heco/info/logo.png" width="32" />         | <https://www.hecochain.com/en-us> |
 | 10001088 | Metis            | METIS  | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/metis/info/logo.png" width="32" />        | <https://www.metis.io/>       |
+| 10001284 | Moonbeam         | GLMR   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/moonbeam/info/logo.png" width="32" />     | <https://moonbeam.network>    |
```
Comment (catenocrypt) [inline docs/registry.md:81]: Logo URL gives 404

File: docs/registry.md [hunk 3]
```diff
@@ -78,6 +78,8 @@ This list is generated from [./registry.json](../registry.json)
...
+| 10001285 | Moonriver        | MOVR   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/moonriver/info/logo.png" width="32" />    | <https://moonbeam.network/networks/moonriver> |
```
Comment (catenocrypt) [inline docs/registry.md:82]: Logo URL gives 404, logo&info has to be added&accepted to assets repo first

**Review by catenocrypt — CHANGES_REQUESTED** [review]
Logos & info is missing in assets.  If PR in assets exists, let me know.

**Comment by catenocrypt** [comment]
> Relevant assets PR: [trustwallet/assets#20876](https://github.com/trustwallet/assets/pull/20876)

Assets PR merged

**Pull request merged**